components/SearchAutocomplete.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  FC,
+  useState,
+  useEffect,
+  useRef,
+  ReactNode,
+  MutableRefObject,
+  RefObject,
+} from 'react'
+
+import {
+  Box,
+  Input,
+  InputProps as IProps,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  useDisclosure,
+  useMultiStyleConfig,
+  InputGroup,
+  InputLeftElement,
+  Text,
+} from '@chakra-ui/react'
+import useOutsideClickHandler from 'hooks/useOutsideClickHandler'
+import { groupBy } from 'ramda'
+
+export type SearchOptionType = {
+  id: string | number
+  value: string | number
+  label: string
+  object: string
+}
+
+interface SearchAutocompleteProps {
+  value?: SearchOptionType
+  InputProps?: IProps
+  options?: SearchOptionType[]
+  emptyOption?: ReactNode
+  renderOption?: (option: SearchOptionType) => ReactNode
+  onSelectOption?: (option: SearchOptionType) => void
+  groupOptionsBy?: (option: SearchOptionType) => string
+  inputLeftElement?: () => ReactNode
+  variant?: string
+  renderGroupTitle?: (title: string) => ReactNode
+}
+
+const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
+  value = null,
+  InputProps = {},
+  options = [],
+  emptyOption = 'No matches',
+  renderOption = option => <Box>{option.label}</Box>,
+  onSelectOption = () => void 0,
+  groupOptionsBy = option => option.object,
+  renderGroupTitle = title => <Text>{title}</Text>,
+  inputLeftElement,
+  ...props
+}) => {
+  const [val, setVal] = useState<SearchOptionType | null>(value)
+  const [search, setSearch] = useState<string>('')
+  const styles = useMultiStyleConfig('SearchAutocomplete', props)
+  const inputRef = useRef<HTMLInputElement>()
+  const popoverRef = useRef<HTMLDivElement>()
+  const { onOpen, onClose, isOpen } = useDisclosure()
+
+  useOutsideClickHandler(
+    [
+      inputRef as MutableRefObject<HTMLElement>,
+      popoverRef as MutableRefObject<HTMLElement>,
+    ],
+    onClose
+  )
+  useEffect(() => {
+    setVal(value)
+  }, [value])
+
+  const groupedOptions = groupBy(groupOptionsBy)(options)
+  const hasOptions = Object.values(groupedOptions).some(
+    groupOptions => groupOptions.length
+  )
+
+  const renderGroup = (groupName: string, groupOptions: SearchOptionType[]) => {
+    return (
+      <>
+        {groupName ? (
+          <Box w="100%" sx={styles?.groupTitleWrapper}>
+            {renderGroupTitle(groupName)}
+          </Box>
+        ) : null}
+        {groupOptions.map((option: SearchOptionType) => (
+          <Box
+            w="100%"
+            key={option.value}
+            sx={styles?.groupOptionWrapper}
+            onClick={() => {
+              if (val !== option) {
+                setVal(option)
+                onSelectOption(option)
+                onClose()
+              }
+            }}
+          >
+            {renderOption(option)}
+          </Box>
+        ))}
+      </>
+    )
+  }
```

Comment:
This feels a lot like a standalone component.

Can you pass the stateful props in and set this up as a `<Group>` component?

Reply:
Also I suggest to use something like next:
```tsx
<Group label="A">
  <Option label="A1"/>
  <Option label="A2"/>
</Group>
<Group label="B">
  <Option label="B1"/>
  <Option label="B2"/>
</Group>
```